### PR TITLE
Feature/user login

### LINKED ISSUE #34: Update/add parameter and function types
Currently, many parameters are untyped and no functions have typed returns.

- [ ] Comb through each component typing functions and untyped parameters in callbacks, etc

### LINKED ISSUE #36: Conditional displays for user
If no user logged in:
- [x] Display `login` button in home menu
- [x] Hide `shelves` button
- [x] Hide `add to shelves` button

if user logged in:
- [x] Display `logout` button in home menu
- [x] Display `shelves` button
- [x] Display `add to shelves` button

### LINKED ISSUE #38: Refactor AlertBar to handle success and error messages
- [x] If success message sent, render success alert
- [x] If failure message sent, render error alert
- [x] Refactor `Nav` component to send success messages dynamically (define a function to return an `AlertBar` element with respective prop depending on operation and status)

### PR DESCRIPTION
**Pull request by tednaphil**
# Livre List FE

## Summary 
- Add/update types for function returns, state variables, parameters, etc closes #34 
- Adds infrastructure for login endpoint (intending to do a hot swap) closes #36 
- `AlertBar` component added to notify user of successful or failed login/logout attempts closes #38 

## Screenshots (if applicable) 
AlertBarPreview

## Changes Made
- `Shelves` and `Add to Shelf` buttons now only visible if a user is "logged in"

## Testing
- Tests added for simulated login

## Checklist
- [ ] The code follows the project's coding standards.
- [x] Tests have been added or updated to cover the changes.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [x] The code compiles without errors.
- [x] All new and existing tests pass.
- [ ] The pull request has been reviewed by at least one other contributor.

## Reviewer Instructions 
- Many of the changes in this PR were adding types to variables, parameters, etc. In the example below, is it conventional/preferable to define the type on both line 12 and line 18? https://github.com/tednaphil/livre-list/blob/a0b71d136c54db9c91c47fc8d29fdc7e26d5ba5f/src/ShelfCtrl/ShelfCtrl.tsx#L12-L21

## Additional Information/Next Steps
- Test assertions for `AlertBar` behavior on login and logout
- [Refactor `Shelves` component](https://github.com/tednaphil/livre-list/issues/42) to avoid unnecessary `GET` requests on sort changes
- Hot swaps for user and shelf endpoints
- CI setup

### CODE REVIEW

File: src/ShelfCtrl/ShelfCtrl.tsx [hunk 1]
```diff
@@ -7,15 +7,15 @@ interface Props {
 }
 
 function ShelfCtrl({ setSort }: Props) {
-  const [newShelf, setNewShelf] = useState('');
+  const [newShelf, setNewShelf] = useState<string>('');
 
-  const handleSort = (e: any,) => {
+  const handleSort = (e: React.ChangeEvent<HTMLSelectElement>): void => {
```
Comment (CassandraGoose) [inline src/ShelfCtrl/ShelfCtrl.tsx:12]: Regarding the question in the PR about needing to explicitly add types in spots like this. The answer is: it depends. 

It depends on whether or not TypeScript has already inferred the type successfully or not. For example, If I remove the types on line 12 and line 18, my linter gets mad on line 12. It says, "Parameter 'e' implicitly has an 'any' type.ts". This means that your gut was right to explicitly type lines 12 and 18, since TS doesn't know what's going on - TS has not inferred the type.

However, the same isn't true of line 18. If I remove the explicit type on line 29, my linter doesn't change - there's no error. If I hover over the 'e' in the callback function, my intellisense in VSCode tells me that it knows what type it is and this means that TS has inferred the type already. In this case, we shouldn't add the type ourselves as we could unknowingly introduce an incorrect type. 

Basically, we should only fix what's broken. 

There are teams out there that insist on manually typing everything, but I've only seen them cause more issues with this. It's also possible that TS is wrong - but that's so rare and if it happens, we should first question if our preceding code is actually accurate. 

Let's chat about some other spots in the code that relate to this concept for some practice: BookProfile, Search, SearchCtrl, etc.
